nautobot/ipam/factory.py

```diff
@@ -303,26 +307,30 @@ class Meta:
     description = factory.Maybe("has_description", factory.Faker("text", max_nb_chars=200), "")
 
     has_location = factory.Faker("pybool")
-    location = factory.Maybe(
-        "has_location", random_instance(lambda: Location.objects.get_for_model(VLANGroup), allow_null=False), None
-    )
+
+    @factory.lazy_attribute
+    def _location(self):
+        eligible_site_list = Site.objects.exclude(
+            pk__in=list(VLANGroup.objects.filter(name=self.name).values_list("site", flat=True))
+        )
```

Comment:
Is this backwards? It's excluding only the site assigned to `self`. If the current object has a site assigned the location  `base_site` has to match. https://github.com/nautobot/nautobot/blob/21c85f25e6458ad97e62484bc89e5effa1328a80/nautobot/ipam/models.py#L1083-L1088

Reply:
The purpose of this query is to make sure that `VLANGroup` does not violate the unique constraint (`name`, `site`). Since we decided to allow globally non-unique names for `VLANGroups`. So what I am doing here is basically query all the `sites` that did not get assigned to the `VLANGroup` with the same name as `self` yet. I then query the `locations` based on the `sites` that are eligible (we do not have a `base_site` filter yet). This way a valid `location` is assigned to the VLANGroup so in case that `has_site` is true, `location.site` wont return a `site` that violates the the unique constraint (`name`, `site`).